src/components/ProtectButtonSelector/index.tsx

```diff
@@ -4,150 +4,181 @@
-                        <hr style={{ padding: 0, margin: 0 }} />
-                        {supportedBuilders.map((builder: string) => <BuilderCheckbox fastMode={fastMode} name={builder} key={builder} selectedBuilders={selectedBuilders} toggleBuilder={toggleBuilder} />)}
-                        <Checkbox label="all" id="all" checked={allBuilders === true || fastMode === true} disabled={fastMode === true} onChange={toggleAllBuilders} />
-                    </div>
-                </SimpleDropdown.HiddenBody>
-            </SimpleDropdown>
-        </GridBlock>
-    );
+  };
+
+  // If the user selects any other hint, the "none" option should be deselected.
+
+  const toggleBuilder = (name: string) => {
+    setBuildersSelection((prevBuilders) => ({
+      ...prevBuilders,
+      [name]: !prevBuilders[name],
+    }));
+  };
+
+  const renderFlashbotsProtectButton = () => (
+    <FlashbotsProtectButton
+      hints={hintsProcessed}
+      builders={selectedBuilders}
+      fast={fastMode}>
+      Connect Wallet to Protect
+    </FlashbotsProtectButton>
+  );
+
+  const renderFastOption = () => (
+    <div className={styles.fastContainer}>
+      <Checkbox
+        label="Fast"
+        id="fast"
+        checked={fastMode === true}
+        onChange={setFastMode}
+      />
+    </div>
+  );
+
+  const renderHintCheckbox = (
+    label: string,
+    checked: boolean,
+    onChange: (val: boolean) => void,
+  ) => (
+    <Checkbox
+      label={label}
+      id={label.toLowerCase()}
+      key={label}
+      checked={checked}
+      onChange={onChange}
+    />
+  );
+
+  const renderMevShareHints = () => (
+    <div>
+      <em>MEV-Share Hints</em>
+      <hr style={{padding: 0, margin: 0}} />
+      <AlignItems horizontal="left">
+        {hintLabels.map((label) =>
+          renderHintCheckbox(label, hints[label] || false, (value) =>
+            setHint(label, value),
+          ),
+        )}
+        {renderHintCheckbox('Hash Only', hashOnly, onSetNoHints)}
+        <div style={{width: 64}} /> {/* spacer */}
+      </AlignItems>
+    </div>
+  );
+
+  const renderBuilders = () => (
+    <div>
+      <em>Builders</em>
+      <hr style={{padding: 0, margin: 0}} />
+      {supportedBuilders.map((builder: string) => (
+        <BuilderCheckbox
+          fastMode={fastMode}
+          name={builder}
+          key={builder}
+          builders={buildersSelection}
+          toggleBuilder={toggleBuilder}
+        />
+      ))}
+    </div>
+  );
+
+  return (
+    <GridBlock>
+      <SimpleDropdown
+        header="Advanced options"
+        onClickHeader={() => {
+          setAdvancedOptionsShown(!advancedOptionsShown);
+        }}
+        isOpen={advancedOptionsShown}>
+        <SimpleDropdown.Body>
+          {renderFlashbotsProtectButton()}
+          {renderFastOption()}
+        </SimpleDropdown.Body>
+        <SimpleDropdown.HiddenBody>
+          {renderMevShareHints()}
+          {renderBuilders()}
```

Comment:
kinda pedantic, but why don't we write these as `FunctionComponent`s? this syntax is just kinda strange

Reply:
you are right. Might as well just drive it to the logical end of refactoring. This more of an intermediary result, where the components are refactored into function calls inside the component definition itself. Inner function definitions make it a bit easier to avoid worrying about props passing. However, it also stops me from making them function components, as it is bad practice to define function components as inner functions, which will lead to new definition each render.